clang/test/Analysis/stack-addr-ps.cpp

```diff
@@ -791,3 +791,58 @@ void global_ptr_to_ptr() {
   *global_pp = nullptr;
 }
 } // namespace leaking_via_indirect_global_invalidated
+
+namespace not_leaking_via_simple_ptr {
+void simple_ptr(const char *p) {
+  char tmp;
+  p = &tmp; // no-warning
+}
+
+void ref_ptr(const char *&p) {
+  char tmp;
+  p = &tmp; // expected-warning{{variable 'tmp' is still referred to by the caller variable 'p'}}
+}
+
+struct S {
+  const char *p;
```

Comment:
How about taking one struct by value holding encapsulating a reference member refering to a stack variable?

Reply:
I do not understand what you mean.

My best interpretation
```C++
struct Ref {
  char &p;
};
void struct_ref(Ref s) {
  char tmp;
  s = {tmp}; // no-warning
}
```
does not compile

Reply:
Ah, yes. Makes sense! All good then.